From c4aa969b57f274354f25d304ff9a71294402fed9 Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Thu, 16 Jan 2014 13:43:29 +0800
Subject: [PATCH 256/280] ENGR00295353 Re-enable BWB for decoders

VPU lib v5.4.23

Re-enable BWB for decoders, because FW bug "FSLARIK-353 [Decoder/All]:
vpu fall in busy loop occasionally when BWB is enabled"
was fixed in FW v3.1.1_r46056.

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 vpu/vpu_lib.c | 3 ---
 vpu/vpu_lib.h | 2 +-
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index c925830..1fb731b 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -2750,9 +2750,6 @@ RetCode vpu_DecOpen(DecHandle * pHandle, DecOpenParam * pop)
 			val |= (pDecInfo->tiledLinearEnable << 11 | 0x03 << 9);
 		val |= 1 << 12;
 
-		/* workaround for BWB blocking issue (ENGR00293425), FW v3.1.1 */
-		val &= ~(1 << 12);
-
 		// workaround for BWB blocking issue (ENGR00231107)
 		if (pDecInfo->openParam.bitstreamFormat == STD_VC1)
 			val &= ~(1 << 12);
diff --git a/vpu/vpu_lib.h b/vpu/vpu_lib.h
index 92489c7..2ad8221 100644
--- a/vpu/vpu_lib.h
+++ b/vpu/vpu_lib.h
@@ -683,7 +683,7 @@ static const SocInfo soc_info[] = {
  * v4.2.2 [2008.09.03] support encoder on MX51
  * v4.0.2 [2008.08.21] add the IOClkGateSet() for power saving.
  */
-#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 4, 22)
+#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 4, 23)
 
 extern unsigned int system_rev;
 
-- 
1.8.0

